### ISSUE TITLE
OS X needs more instructions for building with skia

### ISSUE THREAD

**Issue by tony** [aseprite/aseprite#1081]
Greets,

Following ``INSTALL.md`` @ https://github.com/aseprite/aseprite/blob/8775604e5629bb73d9133e41b46efe3f11d48334/INSTALL.md#mac-os-x-details with the additions. Here is my command:

```
cmake -D "CMAKE_OSX_ARCHITECTURES:STRING=x86_64"  -DUSE_SKIA_ALLEG4=OFF -DUSE_SKIA_BACKEND=ON -DSKIA_DIR=$HOME/study/c++/skia \
-D "CMAKE_OSX_DEPLOYMENT_TARGET:STRING=10.7" \
-D "CMAKE_OSX_SYSROOT:PATH=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk" \
-D "WITH_HarfBuzz:BOOL=OFF" ..
```

Here is what I get:

```
- Building with PNG
-- Creating directory /Users/me/work/c++/aseprite/build/third_party/freetype2/include/freetype/config
-- Creating file /Users/me/work/c++/aseprite/build/third_party/freetype2/include/freetype/config/ftconfig.h
-- Creating file /Users/me/work/c++/aseprite/build/third_party/freetype2/include/freetype/config/ftoption.h
-- Check if the system is big endian
-- Searching 16 bit integer
-- Using unsigned short
-- Check if the system is big endian - little endian
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
SKIA_CORE_LIBRARY
    linked by target "she" in directory /Users/me/work/c++/aseprite/src/she
    linked by target "she" in directory /Users/me/work/c++/aseprite/src/she
SKIA_EFFECTS_LIBRARY
    linked by target "she" in directory /Users/me/work/c++/aseprite/src/she
SKIA_IMAGES_LIBRARY
    linked by target "she" in directory /Users/me/work/c++/aseprite/src/she
SKIA_OPTS_AVX2_LIBRARY
    linked by target "she" in directory /Users/me/work/c++/aseprite/src/she
SKIA_OPTS_AVX_LIBRARY
    linked by target "she" in directory /Users/me/work/c++/aseprite/src/she
SKIA_OPTS_LIBRARY
    linked by target "she" in directory /Users/me/work/c++/aseprite/src/she
SKIA_OPTS_SSE41_LIBRARY
    linked by target "she" in directory /Users/me/work/c++/aseprite/src/she
SKIA_OPTS_SSE42_LIBRARY
    linked by target "she" in directory /Users/me/work/c++/aseprite/src/she
SKIA_OPTS_SSSE3_LIBRARY
    linked by target "she" in directory /Users/me/work/c++/aseprite/src/she
SKIA_PORTS_LIBRARY
    linked by target "she" in directory /Users/me/work/c++/aseprite/src/she
SKIA_SFNT_LIBRARY
    linked by target "she" in directory /Users/me/work/c++/aseprite/src/she
SKIA_UTILS_LIBRARY
    linked by target "she" in directory /Users/me/work/c++/aseprite/src/she

-- Configuring incomplete, errors occurred!
See also "/Users/me/work/c++/aseprite/build/CMakeFiles/CMakeOutput.log".
See also "/Users/me/work/c++/aseprite/build/CMakeFiles/CMakeError.log".
```

So do you have any more notes for how we'd be able to build with skia?

Is skia the kind of thing we can just use as a submodule and build from source like the other external deps? I'm aware its not distributed in packages on distros to my knowledge.

**Comment by tony** [comment]
(Slightly off topic, you mention in https://github.com/aseprite/aseprite/tree/56e663bca437c39e463e21d3bbcc0493bcfbf783/src/she the possibility of SDL2, do you have any opinion on it? I've found it to be quite nice. I've done some simpler stuff with it and can get it to run on OS X, FreeBSD, Ubuntu pretty easily)

**Comment by dacap** [comment]
(same as #1066 and #1074)

Please refer to the dependencies section in the INSTALL.md file: htt0ps://github.com/aseprite/aseprite/blob/master/INSTALL.md#dependencies

Skia is not included in the `third_party` directory and must be compiled separately. It's a huge library with a customized build system (it doesn't use cmake at this moment, but there is some progress in that area).

About SDL, the idea is to have just one back-end with all the instructions to create windows and handle events customized. Generally, these libraries like Allegro, SDL2, SFML, etc. are game oriented, so they don't have stuff to handle file events (dropping a file in the window), or pen pressure, or vertical/horizontal scroll with trackpads (which is a really tricky issue on Windows), etc.

So basically we must write all the code to create a Windows, and we can use Skia to draw stuff in the graphics context.

**Comment by tony** [comment]
@dacap 

Ok

That little snippet ``GYP_DEFINES='skia_gpu=0'`` entails a more effort for a developer to build aseprite than it looks :P. I thought it'd be ok to build skia via CMake, but the only way I've found to build without GPU support requires [grabbing depot tools/gclient](https://skia.org/user/quick/macos).

https://github.com/google/skia/blob/master/site/user/tips.md

**Comment by tony** [comment]
Ok @dacap don't get mad. I gave it more time, and still can't get skia to compile D:  And I'm using the ``chrome/m50`` branch

```
# gclient sync
#  GYP_DEFINES='skia_gpu=0' python bin/sync-and-gyp
#  ninja -C out/Debug

>snip<

../../src/views/mac/SkNSView.mm:397:66: error: use of undeclared identifier 'fGLContext'
    CGLPixelFormatObj format = CGLGetPixelFormat((CGLContextObj)[fGLContext CGLContextObj]);
                                                                 ^
../../src/views/mac/SkNSView.mm:412:6: error: use of undeclared identifier 'fGLContext'
    [fGLContext release];
     ^
../../src/views/mac/SkNSView.mm:413:5: error: use of undeclared identifier 'fGLContext'
    fGLContext = nil;
    ^
../../src/views/mac/SkNSView.mm:417:16: error: use of undeclared identifier 'fGLContext'
    if (nil != fGLContext) {
               ^
../../src/views/mac/SkNSView.mm:418:10: error: use of undeclared identifier 'fGLContext'
        [fGLContext flushBuffer];
         ^
../../src/views/mac/SkNSView.mm:423:9: error: use of undeclared identifier 'fGLContext'
    if (fGLContext) {
        ^
../../src/views/mac/SkNSView.mm:425:45: error: use of undeclared identifier 'fGLContext'
        CGLContextObj ctx = (CGLContextObj)[fGLContext CGLContextObj];
                                            ^
19 errors generated.
[1213/2737] CXX obj/src/animator/animator.SkDisplayMovie.o
ninja: build stopped: subcommand failed.
```

When you build skia, do you have the specific commands you used? Are you on 10.11.4?  Did you put in anything for ``CC``/``CXX``/``CPP``/``CPPFLAGS``?

**Comment by dacap** [comment]
Hmm this is strange. I didn't get those errors on OS X (same options as you). Please try compiling with:

    ninja -C out/Debug dm

And let me know. Other option, I've used `export`:

    export GYP_DEFINES='skia_gpu=0'

But I don't think that affects the `ninja` command.

**Comment by tony** [comment]
Ok the ``ninja -C out/Debug dm`` did it.  I was omitting the ``dm``.

Now, during the last stage of linking aseprite, I get a lot of these:

CMake:

```
❯ cmake -D "CMAKE_OSX_ARCHITECTURES:STRING=x86_64"  -DUSE_SKIA_ALLEG4=OFF -DUSE_SKIA_BACKEND=ON -DSKIA_DIR=$HOME/study/c++/skia -GNinja \
-D "CMAKE_OSX_DEPLOYMENT_TARGET:STRING=10.7" \
-D "CMAKE_OSX_SYSROOT:PATH=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk" \
-D "WITH_HarfBuzz:BOOL=OFF" ..

Users/me/work/c++/aseprite/third_party/zlib/Users/me/work/c++/aseprite/build/third_party/zlib
-- Symbol prefix:
-- curl version=[7.21.6]
CMake Warning (dev) at third_party/curl/CMakeLists.txt:562 (if):
  Policy CMP0054 is not set: Only interpret if() arguments as variables or
  keywords when unquoted.  Run "cmake --help-policy CMP0054" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.

  Quoted variables like "HAVE_FCNTL_O_NONBLOCK" will no longer be
  dereferenced when the policy is set to NEW.  Since the policy is not set
  the OLD behavior will be used.
Call Stack (most recent call first):
CMake Warning (dev) at third_party/curl/CMakeLists.txt:594 (if):
  Policy CMP0054 is not set: Only interpret if() arguments as variables or
  keywords when unquoted.  Run "cmake --help-policy CMP0054" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.

  Quoted variables like "HAVE_GLIBC_STRERROR_R_COMPILE" will no longer be
  dereferenced when the policy is set to NEW.  Since the policy is not set
  the OLD behavior will be used.
Call Stack (most recent call first):
  third_party/curl/CMakeLists.txt:673 (curl_internal_test_run)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at third_party/curl/CMake/CurlCheckCSourceCompiles.cmake:16 (if):
  Policy CMP0054 is not set: Only interpret if() arguments as variables or
  keywords when unquoted.  Run "cmake --help-policy CMP0054" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.

  Quoted variables like "curl_cv_recv" will no longer be dereferenced when
  the policy is set to NEW.  Since the policy is not set the OLD behavior
  will be used.
Call Stack (most recent call first):
  third_party/curl/CMake/OtherTests.cmake:31 (curl_check_c_source_compiles)
  third_party/curl/CMakeLists.txt:792 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at third_party/curl/CMake/CurlCheckCSourceRuns.cmake:16 (if):
  Policy CMP0054 is not set: Only interpret if() arguments as variables or
  keywords when unquoted.  Run "cmake --help-policy CMP0054" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.

  Quoted variables like "HAVE_POLL_FINE" will no longer be dereferenced when
  the policy is set to NEW.  Since the policy is not set the OLD behavior
  will be used.
Call Stack (most recent call first):
  third_party/curl/CMake/OtherTests.cmake:202 (curl_check_c_source_runs)
  third_party/curl/CMakeLists.txt:792 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Building with ZLIB
-- Building with BZip2
-- Building with PNG
-- Creating directory /Users/me/work/c++/aseprite/build/third_party/freetype2/include/freetype/config
-- Creating file /Users/me/work/c++/aseprite/build/third_party/freetype2/include/freetype/config/ftconfig.h
-- Creating file /Users/me/work/c++/aseprite/build/third_party/freetype2/include/freetype/config/ftoption.h
-- Configuring done
CMake Warning (dev):
  Policy CMP0042 is not set: MACOSX_RPATH is enabled by default.  Run "cmake
  --help-policy CMP0042" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

  MACOSX_RPATH is not specified for the following targets:

   zlib

This warning is for project developers.  Use -Wno-dev to suppress it.

-- Generating done
-- Build files have been written to: /Users/me/work/c++/aseprite/build
```

(Note: it doesn't appear to be a zlib/curl issue, I disregarded those in the mean time.)

```
❯ ninja
[44/44] Linking CXX executable bin/aseprite
FAILED: : && /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++   -std=c++11 -stdlib=libc++ -Wall -Wno-switch -std=gnu++11 -O2 -g -DNDEBUG -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/D
eveloper/SDKs/MacOSX10.11.sdk -mmacosx-version-min=10.7 -Wl,-search_paths_first -Wl,-headerpad_max_install_names  src/CMakeFiles/aseprite.dir/main/main.cpp.o  -o bin/aseprite  lib/libapp-lib.a -framework Cocoa -framework Carbon -framework IOKit lib/libcfg-lib.a lib/libcss
-lib.a lib/libfilters-lib.a lib/libflic-lib.a lib/librender-lib.a lib/libdoc-lib.a lib/libscript-lib.a lib/libduktape.a lib/libui-lib.a lib/libshe.a lib/libbase-lib.a lib/libmodp_b64.a /usr/lib/libdl.dylib lib/libgfx-lib.a lib/libfixmath-lib.a /Users/me/study/c++/skia/out
/Release/libskia_core.a /Users/me/study/c++/skia/out/Release/libskia_images.a /Users/me/study/c++/skia/out/Release/libskia_opts.a /Users/me/study/c++/skia/out/Release/libskia_opts_ssse3.a /Users/me/study/c++/skia/out/Release/libskia_opts_sse41.a /Users/me/study/c++/skia/o
ut/Release/libskia_opts_sse42.a /Users/me/study/c++/skia/out/Release/libskia_opts_avx.a /Users/me/study/c++/skia/out/Release/libskia_opts_avx2.a /Users/me/study/c++/skia/out/Release/libskia_ports.a /Users/me/study/c++/skia/out/Release/libskia_effects.a /Users/me/study/c++
/skia/out/Release/libskia_core.a /Users/me/study/c++/skia/out/Release/libskia_images.a /Users/me/study/c++/skia/out/Release/libskia_opts.a /Users/me/study/c++/skia/out/Release/libskia_opts_ssse3.a /Users/me/study/c++/skia/out/Release/libskia_opts_sse41.a /Users/me/study/c
++/skia/out/Release/libskia_opts_sse42.a /Users/me/study/c++/skia/out/Release/libskia_opts_avx.a /Users/me/study/c++/skia/out/Release/libskia_opts_avx2.a /Users/me/study/c++/skia/out/Release/libskia_ports.a /Users/me/study/c++/skia/out/Release/libskia_effects.a lib/libski
a_missing_files.a /Users/me/study/c++/skia/out/Release/libskia_utils.a /Users/me/study/c++/skia/out/Release/libskia_sfnt.a lib/libpixman.a lib/libundo-lib.a lib/libjpeg.a lib/libgiflib.a lib/libpng16.a /usr/lib/libm.dylib lib/libfreetype.a /usr/lib/libbz2.dylib /usr/local
/lib/libpng.dylib lib/libupdater-lib.a lib/libnet-lib.a lib/libcurl.a lib/libz.a -ldl -lldap lib/libtinyxml.a && :
Undefined symbols for architecture x86_64:
  "GrGLSLMulVarBy4f(SkString*, char const*, GrGLSLExpr4 const&)", referenced from:
      GrGLMorphologyEffect::emitCode(GrGLSLFragmentProcessor::EmitArgs&) in libskia_effects.a(effects.SkMorphologyImageFilter.o)
      GrGLLightingEffect::emitCode(GrGLSLFragmentProcessor::EmitArgs&) in libskia_effects.a(effects.SkLightingImageFilter.o)
      GrGLMagnifierEffect::emitCode(GrGLSLFragmentProcessor::EmitArgs&) in libskia_effects.a(effects.SkMagnifierImageFilter.o)
  "GrDeepCopyTexture(GrTexture*, SkBudgeted)", referenced from:
      SkImage::NewFromBitmap(SkBitmap const&) in libskia_core.a(core.SkImage.o)
  "GrResourceKeyHash(unsigned int const*, unsigned long)", referenced from:
      GrRectBlurEffect::CreateBlurProfileTexture(GrTextureProvider*, float) in libskia_effects.a(effects.SkBlurMaskFilter.o)
      GrRRectBlurEffect::Create(GrTextureProvider*, float, SkRRect const&) in libskia_effects.a(effects.SkBlurMaskFilter.o)
      SkColorCubeFilter::asFragmentProcessor(GrContext*) const in libskia_effects.a(effects.SkColorCubeFilter.o)
      GrCircleBlurFragmentProcessor::CreateCircleBlurProfileTexture(GrTextureProvider*, SkRect const&, float, float*) in libskia_effects.a(effects.GrCircleBlurFragmentProcessor.o)

>snip<

      GrConstColorProcessor::GrConstColorProcessor(unsigned int, GrConstColorProcessor::InputMode) in libskia_effects.a(effects.SkXfermodeImageFilter.o)
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for GrSimpleTextureEffect", referenced from:
      SkBitmapProcShader::asFragmentProcessor(GrContext*, SkMatrix const&, SkMatrix const*, SkFilterQuality) const in libskia_core.a(core.SkBitmapProcShader.o)
      SkImageShader::asFragmentProcessor(GrContext*, SkMatrix const&, SkMatrix const*, SkFilterQuality) const in libskia_core.a(core.SkImageShader.o)
      SkBlurMaskFilterImpl::filterMaskGPU(GrTexture*, SkMatrix const&, SkRect const&, GrTexture**, bool) const in libskia_effects.a(effects.SkBlurMaskFilter.o)
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
```

**Comment by dacap** [comment]
It looks like a library was compiled with GPU support. Could you please try to clean the Skia build (`ninja clean`), and compile it again (and maybe reconfigure it again with sync-and-gyp)

**Comment by tony** [comment]
<img width="1136" alt="screen shot 2016-04-25 at 9 41 36 pm" src="https://cloud.githubusercontent.com/assets/26336/14805120/d1f3e6f0-0b2e-11e6-8b54-03084355fe47.png">

Yay. That finally got it. Will do a doc update tonight  / tomorrow.

**Comment by dacap** [comment]
:+1: I'm glad you were able to compiled :wink: looking for your feedback

**Comment by tony** [comment]
@dacap Followed up in #1097

**Comment by locoruiz** [comment]
Hello, I´m having trouble compiling in OS X also.
This is my cmake command:
`cmake -DCMAKE_OSX_ARCHITECTURES=x86_64  -DCMAKE_OSX_DEPLOYMENT_TARGET=10.12 -DCMAKE_OSX_SYSROOT=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk  -DUSE_ALLEG4_BACKEND=OFF  -DUSE_SKIA_BACKEND=ON  -DSKIA_DIR=$HOME/deps/skia -DWITH_HarfBuzz=OFF -G Ninja ..`
And the result at the end is:
```
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
SKIA_OPTS_AVX2_LIBRARY
    linked by target "she" in directory /Users/kawi/Desktop/Aseprite/src/she

-- Configuring incomplete, errors occurred!
See also "/Users/kawi/Desktop/Aseprite/build/CMakeFiles/CMakeOutput.log".
See also "/Users/kawi/Desktop/Aseprite/build/CMakeFiles/CMakeError.log".
```
Any ideas why this is?
Any help would be greatly appreciated.

**Comment by tony** [comment]
I’m having the same issue. I have have retried with all the Skia branches
also.

I think what we really need is to get Travis-CI working so we are so builds
work on Linux and Mac. They’d likely take 60 minutes to finish, but it’s
free and a whole lot better than being surprised by compile errors.


On December 7, 2016 at 10:12:58 AM, locoruiz (notifications@github.com)
wrote:

Hello, I´m having trouble compiling in OS X also.
This is my cmake command:
cmake -DCMAKE_OSX_ARCHITECTURES=x86_64 -DCMAKE_OSX_DEPLOYMENT_TARGET=10.12
-DCMAKE_OSX_SYSROOT=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk
-DUSE_ALLEG4_BACKEND=OFF -DUSE_SKIA_BACKEND=ON -DSKIA_DIR=$HOME/deps/skia
-DWITH_HarfBuzz=OFF -G Ninja ..
And the result at the end is:

CMake Error: The following variables are used in this project, but
they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the
CMake files:
SKIA_OPTS_AVX2_LIBRARY
    linked by target "she" in directory /Users/kawi/Desktop/Aseprite/src/she

-- Configuring incomplete, errors occurred!
See also "/Users/kawi/Desktop/Aseprite/build/CMakeFiles/CMakeOutput.log".
See also "/Users/kawi/Desktop/Aseprite/build/CMakeFiles/CMakeError.log".

Any ideas why this is?
Any help would be greatly appreciated.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/aseprite/aseprite/issues/1081#issuecomment-265490798>,
or mute the thread

**Comment by dacap** [comment]
Hi people, did you try the [aseprite-m55 branch](https://github.com/aseprite/skia/tree/aseprite-m55)? We're using that one for compiling the master version.

**Comment by locoruiz** [comment]
Hello @dacap  I think I did, following the instructions of:
https://github.com/aseprite/aseprite/blob/master/INSTALL.md#skia-on-macos
there is a line that says:
`git checkout aseprite-m55`

**Comment by dacap** [comment]
@locoruiz are you trying to compile master? beta? or v1.1.9? because there is no more `SKIA_OPTS_AVX2_LIBRARY` variable in the she/CMakeLists.txt file

**Comment by locoruiz** [comment]
@dacap I am trying to compile v1.1.9

**Comment by dacap** [comment]
@locoruiz yeah, that's the problem. v1.1.9 requires Skia's [aseprite-m53 branch](https://github.com/aseprite/skia/tree/aseprite-m53). You can follow the INSTALL.md instructions from the v1.1.9 source code, because the INSTALL.md in the master branch is for the master branch (not for v1.1.9). If you've already compiled aseprite-m55 branch, you might try to clone and compile Aseprite's master branch.

**Comment by dacap** [comment]
I'm not sure if that makes sense, but:
```
Aseprite v1.1.9 <-> requires Skia aseprite-m53
Aseprite master <-> requires Skia aseprite-m55
```

**Comment by locoruiz** [comment]
@dacap I thought v1.1.9 was the latest. 
Muchas gracias David, bajaré la versión master e intentaré de nuevo.
Felicidades por el gran trabajo, 
Saludos desde Bolivia

**Comment by dacap** [comment]
@locoruiz v1.1.9 is the latest released version, master (and beta) are the development branches.
Gracias por las palabras! Saludos desde Argentina! 😄

**Comment by lucaslrt** [comment]
I'm trying to install it on Mac OS 14.10.4 and gives the same error:

```
[687/687] Linking CXX executable bin/aseprite
FAILED: bin/aseprite 
: && /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -stdlib=libc++ -Wall -Wno-switch -O2 -g -DNDEBUG -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk -mmacosx-version-min=10.14 -Wl,-search_paths_first -Wl,-headerpad_max_install_names  src/CMakeFiles/aseprite.dir/main/main.cpp.o  -o bin/aseprite  lib/libapp-lib.a lib/libclip.a lib/libdio-lib.a lib/libfilters-lib.a lib/libflic-lib.a lib/librender-lib.a lib/libdoc-lib.a lib/libfixmath-lib.a lib/libui-lib.a lib/liblaf-os.a lib/liblaf-gfx.a lib/libpixman.a lib/liblaf-ft.a -framework Cocoa -framework Carbon /Users/User/deps/skia/out/Release/libskia.a -framework opengl lib/libobs.a lib/libundo.a lib/libcmark.a lib/libjpeg.a lib/libgiflib.a lib/libwebpdemux.a lib/libwebpmux.a lib/libwebp.a lib/libfreetype.a lib/libharfbuzz.a lib/libfreetype.a lib/libharfbuzz.a lib/libpng16.a /usr/lib/libm.dylib /usr/lib/libbz2.dylib lib/libjson11.a lib/libarchive.a /usr/lib/libiconv.dylib /usr/lib/libcharset.dylib /usr/lib/libexpat.dylib lib/libfmt.a lib/libtinyexpr.a lib/liblauxlib.a lib/liblua.a lib/liblualib.a lib/libupdater-lib.a lib/libcfg-lib.a lib/libtinyxml.a lib/libnet-lib.a lib/liblaf-base.a lib/libmodpbase64.a /usr/lib/libdl.dylib lib/libcurl.a lib/libz.a -ldl -lldap && :
Undefined symbols for architecture x86_64:
  "is_orientation_marker(unsigned char const*, unsigned long, SkEncodedOrigin*)", referenced from:
      SkWebpCodec::MakeFromStream(std::__1::unique_ptr<SkStream, std::__1::default_delete<SkStream> >, SkCodec::Result*) in libskia.a(webp.SkWebpCodec.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
```

I've, already tried to compile with other branches on skya and when comes at the aseprite installation it gives other errors before that.
Please, help >.<

**Comment by dacap** [comment]
Hi @lucaslrt! what specific commit are you trying to compile? and what specific Skia version did you compile?

**Comment by lucaslrt** [comment]
aseprite:
```
commit 0264964c4e8ca4b6c7316ee02597c7bae3f2a3c1 (HEAD -> master, origin/master, origin/HEAD)
Author: David Capello <david@igarastudio.com>
Date:   Wed Apr 24 21:14:20 2019 -0300

    lua: Add app.preferences object
```
Skia versions that gives this error: aseprite-m71 / aseprite-m71-tmp

**Comment by lucaslrt** [comment]
So, @dacap, do you know what could be wrong?

**Comment by dacap** [comment]
@lucaslrt not sure, wasn't able to look at this

**Comment by mrspeaker** [comment]
@lucaslrt  - I got this same error as you, fixed it with this: https://github.com/aseprite/aseprite/issues/1981#issuecomment-463119446

I needed to rebuild Skia with `skia_use_libwebp=false` as part of the `gn` command. This gave an warning (about "skia_use_libwebp is not an option") - but still, the build worked correctly after that.
